sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinHelper.scala

```diff
@@ -213,15 +214,39 @@ object StreamingSymmetricHashJoinHelper extends Logging {
       @transient private val storeCoordinator: Option[StateStoreCoordinatorRef])
       extends ZippedPartitionsBaseRDD[V](sc, List(rdd1, rdd2)) {
 
+    lazy private val executorMap: mutable.HashMap[String, Int] = {
```

Comment:
If I understand correctly, we can simply do the round robin and achieve the same thing. 

Initialize the new immutable array or list from `context.getExecutorIds()` (if `context.getExecutorIds()` is also immutable this is also not necessary as well), and have a variable for index while would move by one per assignment (like `++idx % count`).

Reply:
Btw, what we'd like to achieve is trying to distribute state across all executors, then this PR looks to miss accounting executors containing active state. We'll want to also increase the count for the case `stateStoreLocs` is available as well. Probably better than simple round robin.

Reply:
Good point. I will update this.